linera-rpc/src/grpc/node_provider.rs

```diff
@@ -3,37 +3,59 @@
 
 use std::str::FromStr as _;
 
+use linera_base::time::Duration;
 use linera_core::node::{NodeError, ValidatorNodeProvider};
 
 use super::GrpcClient;
-use crate::{config::ValidatorPublicNetworkConfig, node_provider::NodeOptions};
-
-#[derive(Copy, Clone)]
-pub struct GrpcNodeProvider(NodeOptions);
+use crate::{
+    config::ValidatorPublicNetworkConfig,
+    grpc::{pool::GrpcConnectionPool, transport::Options},
```

Comment:
Please refer to this qualified as `transport::Options` since this isn't a transport module.